src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySink.java

```diff
@@ -161,6 +161,12 @@ void initSQLEngineOutput(BatchSinkContext context,
                            String tableName,
                            @Nullable Schema tableSchema,
                            FailureCollector collector) {
+    // Sink Pushdown is not supported if the sink schema is not defined.
+    if (tableSchema == null) {
+      LOG.debug("BigQuery SQL Engine Output was not initialized. Schema was empty.");
+      return;
```

Comment:
so, this just means that bqexecute will not be pushed down for schemaless and will work in the old way, correct?

Reply:
BQ Execute cannot be pushed down either way. Schemaless pipelines still work in Spark just fine.